src/main/java/BallVision.java

```diff
@@ -10,107 +11,234 @@ Basic vision algorithm for detecting balls (powercells)
 import org.opencv.imgproc.*;
 import org.opencv.objdetect.*;
 
-public class BallVision{
+public class BallVision {
 
-//Load 6 test images, process, and write 6 output images. To-Do: Replace with camera input
+    static class MatContourPair {
+        public Mat mat;
+        public List<MatOfPoint> contours;
+
+        public MatContourPair(Mat mat, List<MatOfPoint> contours) {
+            this.mat = mat;
+            this.contours = contours;
+        }
+    }
+
+    private static TargetLocation target;
+
+    //Load test images, process, and write output images. To-Do: Replace with camera input
     public static void main(String[] args) {
         System.loadLibrary(Core.NATIVE_LIBRARY_NAME);
-        for (int i = 1; i<10; i++){
-            String image= "src/main/java/BallImage";
-            image = image + i + ".jpg";
-            Mat input = Imgcodecs.imread(image);
-            Mat output = algorithm(input);
-            String output_fname = "Output Image " + i + ".jpg";
-            Imgcodecs.imwrite(output_fname, output);
+
+        if (args[0] == "TEST") {
+            long hm_images = 18;
+
+            // test using test_inputs, writing to test_outputs
+            for (int i = 1; i <= hm_images; i++) {
+                Mat input = Imgcodecs.imread("src/test_inputs/BallImage" + i + ".jpg");
+                Mat output = decoratePipelineOutput(ballDetectPipeline(input));
+                Imgcodecs.imwrite("src/test_outputs/Output Image " + i + ".jpg",
+                        output);
+            }
+            System.out.println("Done");
+        } else {
+            Mat image = Imgcodecs.imread(args[0]);
+            MatContourPair result = ballDetectPipeline(image);
+
+            double yaw = getTargetYaw(
+                    findBestBallTarget(result.contours),
+                    new Size(result.mat.width(), result.mat.height()),
+                    // the following are copied from HighGoalVision.java
+                    Math.toRadians(68.5),
+                    16.0,
+                    9.0
+            );
```

Comment:
BallVision doesn't need a main method (it is dead code), and it should probably be removed. I know my HighGoalVision includes one (which should probably also be removed).